src/main/scala/dicom/json/DicomJson.scala

```diff
@@ -51,11 +53,59 @@ object DicomJson {
 
   def json2string(json: JsonStructure): String = {
     val sw = new java.io.StringWriter
-    val jwf =
-      javax.json.Json.createWriterFactory(Map.empty.asJava)
+    val jwf = Json.createWriterFactory(Map.empty.asJava)
     val jw = jwf.createWriter(sw)
     jw.write(json)
     jw.close
     sw.toString
   }
+
+  def deepRenameJsonKeys(
+      jsonStruct: JsonStructure,
+      renameF: String => String
+  ): JsonStructure = {
+    jsonStruct match {
+      case jsonArr: JsonArray => {
+        val outBuilder = Json.createArrayBuilder
+        jsonArr.forEach(jsonVal => {
+          val outVal = jsonVal.getValueType match {
+            case JsonValue.ValueType.OBJECT =>
+              deepRenameJsonKeys(jsonVal.asJsonObject, renameF)
+            case JsonValue.ValueType.ARRAY =>
+              deepRenameJsonKeys(jsonVal.asJsonArray, renameF)
+            case _ => jsonVal
+          }
+          outBuilder.add(outVal)
+        })
+        outBuilder.build
+      }
+      case jsonObj: JsonObject => {
+        val outBuilder = Json.createObjectBuilder
+        jsonObj.entrySet.asScala.foreach(entry => {
+          val outKey = renameF(entry.getKey)
+          val outValue = entry.getValue.getValueType match {
+            case JsonValue.ValueType.OBJECT =>
+              deepRenameJsonKeys(entry.getValue.asJsonObject, renameF)
+            case JsonValue.ValueType.ARRAY =>
+              deepRenameJsonKeys(entry.getValue.asJsonArray, renameF)
+            case _ => entry.getValue
+          }
+          outBuilder.add(outKey, outValue)
+        })
+        outBuilder.build
+      }
+    }
+  }
```

Comment:
It seems to me that you can refactor it, but it's probably okay to let it like this since it's rather small